src/components/ButtonGroupPicker/index.js

```diff
@@ -0,0 +1,145 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import StyledContainer from './styled/container';
+import StyledButtonGroup from './styled/buttonGroup';
+import StyledLegend from './styled/label';
+import { uniqueId } from '../../libs/utils';
+import { Provider } from './context';
+import RenderIf from '../RenderIf';
+import HelpText from './styled/helpText';
+import ErrorText from './styled/errorText';
+
+/**
+ * ButtonGroupPicker can be used to group related options. The ButtonGroupPicker will control the selected state of its child ButtonOption.
+ */
+class ButtonGroupPicker extends Component {
+    constructor(props) {
+        super(props);
+        this.groupNameId = this.props.name || uniqueId('options');
+        this.errorMessageId = uniqueId('error-message');
+        this.handleOnChange = this.handleOnChange.bind(this);
+    }
+
+    getValue() {
+        const { value } = this.props;
+        if (typeof value === 'string') {
+            return [];
+        }
+        return value;
+    }
+
+    getErrorMessageId() {
+        const { error } = this.props;
+        if (error) {
+            return this.errorMessageId;
+        }
+        return undefined;
+    }
+
+    handleOnChange(event) {
+        const { value, checked } = event.target;
+        const { value: values, multiple, onChange } = this.props;
+
+        if (!multiple) {
+            return onChange([value]);
+        }
+
+        if (checked && Array.isArray(values)) {
+            return onChange(values.concat([value]));
+        }
+        if (checked && !Array.isArray(values)) {
+            return onChange([value]);
+        }
+        return onChange(values.filter(valueId => valueId !== value));
+    }
+
+    render() {
+        const {
+            name,
+            className,
+            style,
+            label,
+            multiple,
+            children,
+            size,
+            error,
+            bottomHelpText,
+        } = this.props;
+
+        const context = {
+            onChange: this.handleOnChange,
+            values: this.getValue(),
+            type: multiple ? 'checkbox' : 'radio',
+            name: name || uniqueId('option'),
+            ariaDescribedBy: this.getErrorMessageId(),
```

Comment:
in order to avoid calling getErrorMessageId two times in each rerender you can put it in a constant to use here and in the ErrorText